bluebrain/repo-patches/packages/neuron/package.py

```diff
@@ -216,6 +217,9 @@ def cmake_enable_option(spec_requiremement):
                 "+tests",
             ]
         ]
+        if self.spec.satisfies("+backtrace"):
+            args.append(self.define("NRN_ENABLE_BACKTRACE", True))
```

Comment:
LGTM but two questions:
* should we make the backtrace ON by default?
* in our production deployment if we want nicer backtraces (which is desirable), should add libbfd/libdwarf like dependency when `+backtrace`?

Reply:
It's been off. Hence to avoid any breakage, I left it off. As for nicer backtraces, those require the source code.  I think we build on NVME drives and then delete the sources. In that case the output is:
```
$ nrniv raise.py
NEURON -- VERSION 9.0a-35-gc583914c8+ 1uc/signal (c583914c8+) 2023-09-14
Duke, Yale, and the BlueBrain Project -- Copyright 1984-2022
See http://neuron.yale.edu/neuron/credits

Stack trace (most recent call last):
#20   Object "/usr/lib64/ld-2.17.so", at 0xffffffffffffffff, in 
#19   Object "/gpfs/bbp.cscs.ch/home/groshein/spack_install/software/install_oneapi-2022.2.1-skylake/neuron-develop-nzhp7c/bin/nrniv", at 0x403465, in _start
#18   Object "/usr/lib64/libc-2.17.so", at 0x7fffec46e554, in __libc_start_main
#17   Source "/gpfs/bbp.cscs.ch/home/groshein/nrn-backtrace/src/ivoc/nrnmain.cpp", line 71, in nrnmpi_init [0x4035fd]
#16   Source "/gpfs/bbp.cscs.ch/home/groshein/nrn-backtrace/src/ivoc/ivocmain.cpp", line 757, in ivocmain_session [0x7fffed5ca2ef]
#15   Source "/gpfs/bbp.cscs.ch/home/groshein/nrn-backtrace/src/oc/hoc.cpp", line 846, in hoc_main1 [0x7fffed707fce]
#14   Source "/gpfs/bbp.cscs.ch/home/groshein/nrn-backtrace/src/oc/hoc.cpp", line 1060, in hoc_execute [0x7fffed7085bd]
#13   Source "/gpfs/bbp.cscs.ch/home/groshein/nrn-backtrace/src/nrnpython/nrnpython.cpp", line 125, in nrnpy_pyrun [0x7fffed7bb6ef]
#12   Source "Python/pythonrun.c", line 116, in PyRun_AnyFileExFlags [0x7fffed160558]
#11   Source "Python/pythonrun.c", line 90, in _PyRun_AnyFileObject [0x7fffed16049b]
#10 | Source "Python/pythonrun.c", line 456, in pyrun_file
      Source "Python/pythonrun.c", line 1208, in _PyRun_SimpleFileObject [0x7fffed15fe53]
#9  | Source "Python/pythonrun.c", line 1312, in run_eval_code_obj
      Source "Python/pythonrun.c", line 1291, in run_mod [0x7fffed15ed18]
#8  | Source "Python/ceval.c", line 1134, in _PyEval_Vector
    | Source "Python/ceval.c", line 5065, in _PyEval_EvalFrame
      Source "./Include/internal/pycore_ceval.h", line 46, in PyEval_EvalCode [0x7fffed11a9bc]
#7  | Source "Python/ceval.c", line 4181, in call_function
    | Source "Python/ceval.c", line 5891, in PyObject_Vectorcall
    | Source "./Include/cpython/abstract.h", line 123, in _PyObject_VectorcallTstate
    | Source "./Include/cpython/abstract.h", line 99, in _PyObject_VectorcallTstate
      Source "./Include/cpython/abstract.h", line 112, in _PyEval_EvalFrameDefault [0x7fffecfcee83]
#6    Source "Objects/call.c", line 215, in _PyObject_MakeTpCall [0x7fffed0252df]
#5    Source "/gpfs/bbp.cscs.ch/home/groshein/nrn-backtrace/src/nrnpython/nrnpy_hoc.cpp", line 760, in hocobj_call [0x7fffed7bed5e]
#4    Source "/gpfs/bbp.cscs.ch/home/groshein/nrn-backtrace/src/nrniv/../ivoc/ocjump.cpp", line 138, in fpycall [0x7fffed699b8e]
#3    Source "/gpfs/bbp.cscs.ch/home/groshein/nrn-backtrace/src/nrnpython/nrnpy_hoc.cpp", line 692, in hoc_push [0x7fffed7bf501]
#2    Source "/gpfs/bbp.cscs.ch/home/groshein/nrn-backtrace/src/oc/code.cpp", line 1437, in hoc_errno_check [0x7fffed6f6f55]
#1    Source "/gpfs/bbp.cscs.ch/home/groshein/nrn-backtrace/src/nrnoc/container.cpp", line 164, in hoc_pushx [0x7fffed6bfc2a]
#0    Object "/usr/lib64/libpthread-2.17.so", at 0x7fffec8294fb, in raise
Segmentation fault (Signal sent by tkill() [0x3e8f4000033db])
Segmentation fault
```
even with `libdwarf`. Which is progress since it includes line numbers. When combining with HPE MPI the output will be interleaved.

Reply:
Final consideration, the CMake option `-DNRN_ENABLE_BACKTRACE` defaults to `OFF`.